Chicken.pde

```diff
@@ -0,0 +1,84 @@
+enum ChickenState {
+  IDLE, EXPLORING, FLEEING
+};
+
+class Chicken {
+
+  PVector pos, winSize, target, moveDir = new PVector(0, 0);
+  float moveSpeed = 0f;
+  int ticSpeed, ticCounter = 0;
+
+  ChickenState state = ChickenState.IDLE;
+
+  Chicken(PVector pWindowSize, float pMoveSpeed) {
+    winSize = new PVector(pWindowSize.x, pWindowSize.y);
+    pos = new PVector(winSize.x*0.5f, winSize.y *0.5f);
+    moveSpeed = pMoveSpeed;
+  }
+
+
+  void move() {
+    PVector dir = new PVector(moveDir.x, moveDir.y); //Needed to not keep multiplying moveDir
```

Comment:
You can use `moveDir.copy()` for this, too, by the way.

Reply:
Or you could even do `pos.add(PVector.mult(dir, moveSpeed));` to condense the two statements into one.